### PR TITLE
'const' input support for SYCL-DNN

### PR DESCRIPTION
**Pull request by muhammad-tanvir-1211**

### CODE REVIEW

**Review by KumudhaN — APPROVED** [review]
Looks good to me.

**Review by Bensuo — CHANGES_REQUESTED** [review]
LGTM, but please document this new CMake option in `README.md`

File: cmake/CmakeFunctionHelper.cmake [hunk 1]
```diff
@@ -244,14 +244,24 @@ endfunction(generate_blas_binary_special_objects)
 # blas ternary function for generating source code
 function(generate_blas_ternary_objects blas_level func)
 set(LOCATION "${SYCLBLAS_GENERATED_SRC}/${blas_level}/${func}/")
+string(FIND ${func} "_const" pos)
```
Comment (ProGTX) [inline cmake/CmakeFunctionHelper.cmake:247]:
```suggestion
string(FIND ${func} "_const" const_pos)
```
Or something like that, to make it more descriptive

**Review by ProGTX — CHANGES_REQUESTED** [review]

File: cmake/CmakeFunctionHelper.cmake [hunk 2]
```diff
@@ -244,14 +244,24 @@ endfunction(generate_blas_binary_special_objects)
...
+if(pos)
```
Comment (mehdi-goli) [inline cmake/CmakeFunctionHelper.cmake:248]:
```suggestion
if(const_pos)
```

File: cmake/CmakeFunctionHelper.cmake [hunk 3]
```diff
@@ -244,14 +244,24 @@ endfunction(generate_blas_binary_special_objects)
...
+  string(REPLACE "_const" "" actualfunc ${func})
+endif()
 foreach(executor ${executor_list})
   foreach(data ${data_list})
     cpp_type(cpp_data ${data})
-    set(container_list "BufferIterator<${cpp_data},codeplay_policy>")
+    set(container_list_in)
+    if(pos EQUAL -1)
```
Comment (mehdi-goli) [inline cmake/CmakeFunctionHelper.cmake:255]:
```suggestion
    if(const_pos EQUAL -1)
```

**Comment by KumudhaN** [comment]
@Bensuo  and @ProGTX. I think all the comments were resolved. Can you approve the PR?

**Comment by mehdi-goli** [comment]
> @Bensuo and @ProGTX. I think all the comments were resolved. Can you approve the PR?

@KumudhaN please see [this](https://github.com/codeplaysoftware/sycl-blas/pull/306#pullrequestreview-759615731) comment. It is important to have the readme documentation for each option we are putting, so that we know, when, how and why to use it.

**Comment by KumudhaN** [comment]
> @KumudhaN please see [this](https://github.com/codeplaysoftware/sycl-blas/pull/306#pullrequestreview-759615731) comment. It is important to have the readme documentation for each option we are putting, so that we know, when, how and why to use it.

Thanks, I had missed this. This is fixed now.

**Review by Bensuo — APPROVED** [review]

**Review by ProGTX — APPROVED** [review]

**Pull request merged**